src/components/AiChat/AiChat.tsx

```diff
@@ -21,16 +21,78 @@ import EllipsisIcon from "./EllipsisIcon"
 import { SimpleSelectField } from "../SimpleSelect/SimpleSelect"
 import { useFetch } from "./utils"
 import { SelectChangeEvent } from "@mui/material/Select"
-import { MathJaxContext } from "better-react-mathjax"
+import { type MathJax3Config, MathJaxContext } from "better-react-mathjax"
+import deepmerge from "@mui/utils/deepmerge"
 
 const ConditionalMathJaxWrapper: React.FC<{
   useMathJax: boolean
+  config?: MathJax3Config
   children: React.ReactNode
-}> = ({ useMathJax, children }) => {
+}> = ({ useMathJax, config = {}, children }) => {
   if (!useMathJax) {
     return <>{children}</>
   }
-  return <MathJaxContext>{children}</MathJaxContext>
+  return (
+    <MathJaxContext
+      config={deepmerge(
+        {
+          startup: {
+            typeset: false,
+          },
+          loader: { load: ["[tex]/boldsymbol"] },
+          tex: {
+            packages: { "[+]": ["boldsymbol"] },
+          },
+        },
+        config,
+      )}
+    >
+      {children}
+    </MathJaxContext>
+  )
+}
+
+/**
+ * Component that provides isolation between React and MathJax DOM manipulation
+ *
+ * Seeing errors e.g. Error: Failed to execute 'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
+ *
+ * MathJax manipulates the DOM directly, and when React tries to reconcile during updates during streaming, it encounters DOM nodes that MathJax has modified or replaced.
+ *
+ * Here we hash the content to provide a key to ensure React creates new DOM elements when the content changes instead of trying to reconcile with MathJax modifications.
+ */
+const MathJaxSafeMessage: React.FC<{
+  message: Message
+  useMathJax: boolean
+  className: string
+}> = ({ message, useMathJax, className }) => {
+  const containerRef = useRef<HTMLDivElement>(null)
+
+  const messageContent =
+    typeof message.content === "string" ? message.content : ""
+
+  const contentKey = useMemo(() => {
+    const hash = messageContent.slice(0, 100) + messageContent.length
```

Comment:
AiDrawerManager has a `hashPayload` function. Would you rather put that in a util and use it here, too?